### PR TITLE
feat: branch protection rules

### PR DESCRIPTION
**Pull request by kkania-splunk**
PR for setting branch protection rules in TA repositories
new job `branch-protection-rules` is added, it creates branch protection rules where are not present and updates where are already exist
some explicite given repositories (cyberark-epm) and those where `pre-publish` job were not merged should be skipped

### CODE REVIEW

File: tools/branch_protection_rules.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            requiresCommitSignatures
+            requiresLinearHistory
+            isAdminEnforced
+            allowsForcePushes
+            allowsDeletions
+        }
+    }
+}"
+
+update_pattern="${create_pattern/repositoryId/branchProtectionRuleId}"
```
Comment (arys-splunk) [inline tools/branch_protection_rules.sh:57]: Just to double check - it is fine that there 2 `update_pattern` lines here?

File: tools/branch_protection_rules.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    organization(login: \""$REPOORG"\") {
+        teams(first: 1, query: \""$ADMIN_TEAM"\") {
+            totalCount
+            nodes { name id }
+        }
+    }
+}
+"
+
+admin_team_id=$(gh api graphql -F "query=$look_for_admin_team_id" -q ".data.organization.teams.nodes | .[] | .id")
```
Comment (arys-splunk) [inline tools/branch_protection_rules.sh:70]: What do you think about storing it static, not making a request each time?
I assume that value should not change.
It that is confidential, we can store it in secrets.

File: .github/workflows/sync.yaml [hunk 3]
```diff
@@ -40,11 +40,29 @@ jobs:
...
           git config --global user.name ${{ secrets.SA_GH_USER_ADMIN }}
           git config --global commit.gpgsign true
-          python3 tools/sync.py
+          python3 tools/sync.py tools/sync.sh repositories_main.csv
         env:
           GH_USER_ADMIN: ${{ secrets.SA_GH_USER_ADMIN }}
           GH_TOKEN_ADMIN: ${{ secrets.SA_GH_TOKEN_ADMIN }}
           GH_USER_EMAIL: ${{ secrets.SA_GH_USER_EMAIL }}
+
+  branch-protection-rules:
```
Comment (arys-splunk) [inline .github/workflows/sync.yaml:49]: What is the condition we need to sync branch protection rules?
Only if `tools/branch_protection_rules.sh`,  `repositories_main.csv` are changed?

Comment (arys-splunk) [reply .github/workflows/sync.yaml:49]: It condition is simple as that, then we can use https://github.com/dorny/paths-filter to only run branch protection rules, if it is actually needed.

File: tools/branch_protection_rules.sh (hunk 1 above)
Comment (kkania-splunk) [reply tools/branch_protection_rules.sh:57]: yes, it's intentional, we need those 2 changes to make `update_pattern`

File: .github/workflows/sync.yaml (hunk 3 above)
Comment (uoboda-splunk) [reply .github/workflows/sync.yaml:49]: we can also add `.github/workflows/sync.yaml` to these trigger files

Comment (kkania-splunk) [reply .github/workflows/sync.yaml:49]: very good point, added

File: tools/branch_protection_rules.sh (hunk 2 above)
Comment (kkania-splunk) [reply tools/branch_protection_rules.sh]: we moved it to secrets

**Review by uoboda-splunk — APPROVED** [review]

**Comment by uoboda-splunk** [comment]
LGTM

**Pull request merged**